### PR TITLE
Add configuration flag for  #141

### PR DESCRIPTION
**Pull request by ingwarr**
added enable_binary_authorization flag

### CODE REVIEW

**Comment by alexkonkin** [comment]
> 
> 
> added enable_binary_authorization flag

Your lint tests failed because you should execute 2 commands before commit:
make generate
make generate_docs

**Comment by alexkonkin** [comment]
Please also don't forget to add the tests that validate that your feature is present in the newly created cluster.

**Comment by ingwarr** [comment]
@alexkonkin Thank you!

**Comment by ingwarr** [comment]
> > added enable_binary_authorization flag
> 
> Your lint tests failed because you should have executed 2 commands before commit:
> make generate
> make generate_docs
Fixed, I've added appropriate variables to autogen

File: autogen/cluster_regional.tf [hunk 1]
```diff
@@ -36,6 +36,9 @@ resource "google_container_cluster" "primary" {
   logging_service    = "${var.logging_service}"
   monitoring_service = "${var.monitoring_service}"
 
+{% if private_cluster %}
```
Comment (aaron-lane) [inline autogen/cluster_regional.tf]: What is the motivation to only provide this for private clusters?

File: autogen/outputs.tf [hunk 2]
```diff
@@ -78,6 +78,13 @@ output "ca_certificate" {
   value       = "${local.cluster_ca_certificate}"
 }
 
+{% if private_cluster %}
+output "enable_binary_authorization" {
+description = "Enable BinAuthZ Admission controller"
+value = "${var.enable_binary_authorization}"
```
Comment (aaron-lane) [inline autogen/outputs.tf]: Generally, we do not need to output variables unless there are no other meaningful outputs of a module. This should be dropped.

File: autogen/variables.tf [hunk 3]
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
   default = []
 }
 
+{% if private_cluster %}
+variable "enable_binary_authorization" {
+description = "Enable BinAuthZ Admission controller"
+default     = false
```
Comment (aaron-lane) [inline autogen/variables.tf]:
```suggestion
  default     = false
```

File: autogen/variables.tf [hunk 4]
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
   default = []
 }
 
+{% if private_cluster %}
+variable "enable_binary_authorization" {
+description = "Enable BinAuthZ Admission controller"
```
Comment (aaron-lane) [inline autogen/variables.tf]:
```suggestion
  description = "Enable BinAuthZ Admission controller"
```

File: examples/deploy_service/README.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ It will:
 |------|-------------|:----:|:-----:|:-----:|
 | cluster\_name\_suffix | A suffix to append to the default cluster name | string | `""` | no |
 | compute\_engine\_service\_account | Service account to associate to the nodes in the cluster | string | n/a | yes |
+| enable\_binary\_authorization | Enable BinAuthZ Admission controller | string | `"false"` | no |
```
Comment (aaron-lane) [inline examples/deploy_service/README.md]: These tables should be edited only through `make generate_docs`, not manually; this variable has not been added to any of the examples.

```suggestion
```

File: autogen/outputs.tf (hunk 2 above)
Comment (ingwarr) [reply autogen/outputs.tf]: Ok, I'll do it

Comment (ingwarr) [reply autogen/outputs.tf]: Done

File: autogen/cluster_regional.tf (hunk 1 above)
Comment (ingwarr) [reply autogen/cluster_regional.tf]: Because of this change valid only for google-beta provider

Comment (aaron-lane) [reply autogen/cluster_regional.tf]: Got it. Thanks!

File: autogen/outputs.tf [hunk 6]
```diff
@@ -78,6 +78,7 @@ output "ca_certificate" {
   value       = "${local.cluster_ca_certificate}"
 }
 
+
```
Comment (aaron-lane) [inline autogen/outputs.tf]:
```suggestion
```

File: autogen/variables.tf [hunk 7]
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
   default = []
 }
 
+{% if private_cluster %}
+variable "enable_binary_authorization" {
+  description = "Enable BinAuthZ Admission controller"
+  default     = false
+}
+{% endif %}
+
```
Comment (aaron-lane) [inline autogen/variables.tf]:
```suggestion
```

File: autogen/variables.tf [hunk 8]
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
   default = []
 }
 
+{% if private_cluster %}
+variable "enable_binary_authorization" {
+  description = "Enable BinAuthZ Admission controller"
+  default     = false
+}
```
Comment (aaron-lane) [inline autogen/variables.tf]:
```suggestion
}

```

File: modules/private-cluster/outputs.tf [hunk 9]
```diff
@@ -78,6 +78,11 @@ output "ca_certificate" {
   value       = "${local.cluster_ca_certificate}"
 }
 
+output "enable_binary_authorization" {
```
Comment (aaron-lane) [inline modules/private-cluster/outputs.tf]: Please remove this output; it doesn't provide any value.

Comment (ingwarr) [reply modules/private-cluster/outputs.tf]: Ok, done

**Comment by alexkonkin** [comment]
Quoting the results of the integration-tests execution:
a)
google_compute_subnetwork.main: Error waiting to create Subnetwork: Error waiting for Creating Subnetwork: Quota 'ROUTES' exceeded.  Limit: 250.0 globally.
b)
Error waiting for Deleting Network: The network resource '<...>/networks/cft-gke-test-5pwt' is already being used by 'projects/ci-kubernetes-engine/regions/us-east4/subnetworks/cft-gke-test-5pwt'

It seems that some resources have not been deleted after the previous CI execution....
As far as I remember such situation can be resolved by the one who has
the administrator's permission to do this.

**Comment by aaron-lane** [comment]
I manually verified the node pool suite; the failure is unrelated to these changes.

**Pull request merged**